### PR TITLE
Add support for new and existing groups for iTC testers in spaceship

### PR DESCRIPTION
**Pull request by KrauseFx**
Until now we haven't really supported groups in _spaceship_, this PR adds full support for existing group IDs, group names, and creating new groups on the fly 👍

### CODE REVIEW

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 1]
```diff
@@ -881,21 +885,42 @@ def create_tester!(tester: nil, email: nil, first_name: nil, last_name: nil, gro
...
+          value: last_name || ""
+        },
+        testing: {
+          value: true
+        }
+      }
+
+      if groups
+        tester_data[:groups] = groups.map do |group_name_or_group_id|
+          if self.groups.values.include?(group_name_or_group_id)
```
Comment (mfurtak) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: `self.groups.has_value? (group_name_or_group_id)`

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 2]
```diff
@@ -881,21 +885,42 @@ def create_tester!(tester: nil, email: nil, first_name: nil, last_name: nil, gro
...
+            # This is an existing group, let's use that, the user specified the group name
+            group_name = group_name_or_group_id
+            group_id = self.groups.find { |id, name| name == group_name_or_group_id }[0]
```
Comment (mfurtak) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: `self.groups.key(group_name_or_group_id)`

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 3]
```diff
@@ -881,21 +885,42 @@ def create_tester!(tester: nil, email: nil, first_name: nil, last_name: nil, gro
...
+          elsif self.groups.keys.include?(group_name_or_group_id)
```
Comment (mfurtak) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: `self.groups.has_key?(group_name_or_group_id)`

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 4]
```diff
@@ -881,21 +885,42 @@ def create_tester!(tester: nil, email: nil, first_name: nil, last_name: nil, gro
...
+            # This is an existing group, let's use that, the user specified the group ID
+            group_name = self.groups.find { |id, name| id == group_name_or_group_id }[1]
```
Comment (mfurtak) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: `self.groups[group_name_or_group_id]`

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 5]
```diff
@@ -881,21 +885,42 @@ def create_tester!(tester: nil, email: nil, first_name: nil, last_name: nil, gro
...
+            group_id = group_name_or_group_id
+          else
+            group_name = group_name_or_group_id
+            group_id = nil # this is expected by the iTC API
+          end
+
+          {
+            "id" => group_id,
+            "name" => {
+              "value" => group_name
```
Comment (mfurtak) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: IMO this block is a lot for the contents of a `map`. Consider turning this into its own method and call that during the `map`

File: spaceship/spec/tunes/testers_spec.rb [hunk 6]
```diff
@@ -70,6 +70,64 @@
...
+        groups: ["boarding123", "b6f65dbd-c845-4d91-bc39-0b661d608970", "custom-new-group"]
+      )
+
+      expect(tester.first_name).to eq(first_name)
+      expect(tester.last_name).to eq(last_name)
+      expect(tester.groups.count).to eq(3)
+      expect(tester.groups.first["id"]).to eq("c4739ccc-672f-46d4-9dad-23df8c60a35f")
+      expect(tester.groups.first["name"]["value"]).to eq("boarding123")
+    end
+  end
```
Comment (mfurtak) [inline spaceship/spec/tunes/testers_spec.rb]: Nice!

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 2 above)
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: omg 😮

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 4 above)
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Yah, I wasn't thinking on this one

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 5 above)
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Not a big fan of adding a new method and clutter the class' methods scope for this, as we won't need to be calling it from anywhere else. Is there any better way?

Comment (mfurtak) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Could make a new class/file and put the helper method in there, I guess? If you don't want to pull it out, I understand.

**Review by mfurtak — APPROVED** [review]
Good to go when you're ready 👍

**Pull request merged**